app/components/file-uploader.js

```diff
@@ -151,99 +155,78 @@ export default Component.extend(Analytics, {
-                    this.set('newNode', true);
-                    this.set('applyLicense', true);
-                })
-                .catch(() => {
-                    this.set('uploadInProgress', false);
-                    this.get('toast').error(this.get('i18n').t('components.file-uploader.could_not_create_component'));
-                });
+                .then(this._createComponent.bind(this))
+                .catch(this._failCreateComponent.bind(this));
         },
 
         uploadFileToExistingNode() {
-            // Upload case for using an existing node with a new file for the preprint.  Updates title of existing node and then uploads file to node.
+            // Upload case for using an existing node with a new file for the preprint.
+            // Updates title of existing node and then uploads file to node.
             // Also applicable in edit mode.
             this.get('metrics')
                 .trackEvent({
                     category: 'button',
                     action: 'click',
-                    label: `${this.get('editMode') ? 'Edit' : 'Submit'} - Save and Continue, ${this.get('nodeLocked') ? 'Save File/Title Edits' : 'Existing Node New File'}`
+                    label: `${this.get('editMode') ? 'Edit' : 'Submit'} - Save and Continue, ${this.get('nodeLocked') ? 'Save File/Title Edits' : 'Existing Node New File'}`,
                 });
             if (this.get('nodeLocked')) { // Edit mode
                 this.set('uploadInProgress', true);
             }
 
-            let model = this.get('model');
-            let node = this.get('node');
+            const model = this.get('model');
+            const node = this.get('node');
             this.set('basicsAbstract', this.get('model.description') || null);
-            let currentNodeTitle = node.get('title');
+            const currentNodeTitle = node.get('title');
+            this.set('currentNodeTitle', currentNodeTitle);
 
             if (currentNodeTitle !== this.get('title')) {
                 model.set('title', this.get('title'));
                 node.set('title', this.get('title'));
                 node.save()
-                .then(() => {
-                    this.send('upload');
-                })
-                .catch(() => {
-                    node.set('title', currentNodeTitle);
-                    this.set('uploadInProgress', false);
-                    this.get('toast').error(this.get('i18n').t('components.file-uploader.could_not_update_title'));
-                });
+                    .then(this._sendToUpload.bind(this))
+                    .catch(this._failUpdateTitle.bind(this));
             } else {
                 this.send('upload');
             }
         },
 
         // Dropzone hooks
         sending(_, dropzone, file, xhr/* formData */) {
-            let _send = xhr.send;
+            const _send = xhr.send;
+            /* eslint-disable-next-line no-param-reassign */
```

Comment:
probably could be refactored, but wasn't a high priority since it is pretty clear what is going on

Reply:
It's complaining because we are passing in `xhr` and then assigning `xhr.send`. It looks like we are overriding `xhr.send` on purpose but I haven't looked to see how it is being used. Might come back to this one.